### PR TITLE
Pvayladev 799 more fail safe sensor registery

### PR DESCRIPTION
**Pull request by Jevli**

### CODE REVIEW

File: src/monitor/src/main/java/ee/ria/xroad/monitor/MetricRegistryHolder.java [hunk 1]
```diff
@@ -22,13 +22,19 @@
...
 
+import com.codahale.metrics.Histogram;
 import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SlidingTimeWindowReservoir;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Global access point for {@link MetricRegistry}
```
Comment (jansu76) [inline src/monitor/src/main/java/ee/ria/xroad/monitor/MetricRegistryHolder.java]: getOrCreateXXX methods are not thread safe, if multiple concurrent threads use them with same metricName, dangling SimpleSensors/Histogram objects are created. Either add comments about thread safety, or implement locking to achieve thread safety.

File: src/monitor/src/main/java/ee/ria/xroad/monitor/CertificateInfoSensor.java [hunk 2]
```diff
@@ -85,7 +83,6 @@ public CertificateInfoSensor() throws Exception {
                 .addExtractor(CertificateType.SECURITY_SERVER_TLS, new InternalTlsExtractor())
                 .addExtractor(CertificateType.AUTH_OR_SIGN, new TokenExtractor());
 
-        updateOrRegisterData(new JmxStringifiedData());
```
Comment (jansu76) [inline src/monitor/src/main/java/ee/ria/xroad/monitor/CertificateInfoSensor.java]: why was this removed, and does it alter behaviour if metrics data is fetched before INITIAL_DELAY?

File: src/monitor/src/main/java/ee/ria/xroad/monitor/MetricRegistryHolder.java (hunk 1 above)
Comment (Jevli) [reply src/monitor/src/main/java/ee/ria/xroad/monitor/MetricRegistryHolder.java]: Now they are thread safe!

File: src/monitor/src/main/java/ee/ria/xroad/monitor/CertificateInfoSensor.java (hunk 2 above)
Comment (Jevli) [reply src/monitor/src/main/java/ee/ria/xroad/monitor/CertificateInfoSensor.java]: I didn't return "empty" initialisations but for other two sensor they are initialised. Also tested that update missing from constructor doesn't break monitor request.
